### PR TITLE
Support for null date states

### PR DESCRIPTION
**Pull request by hahahana**
This PR adds the ability to initialize a datepicker with a null value. This also introduces placeholder text.

You can test this [live](http://cdn.rawgit.com/hahahana/react-datepicker/21464e529b72d17a2075ea55c26eb449c0d3d2fe/example.html).

----
This should close https://github.com/Hacker0x01/react-datepicker/issues/53

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Would it be worth to add some tests that support this feature?

File: react-datepicker.js [hunk 1]
```diff
@@ -156,29 +156,29 @@ var Calendar = React.createClass({displayName: "Calendar",
 
   render: function() {
     return (
-      React.createElement("div", {className: "datepicker"},
```
Comment (martijnrusschen) [inline react-datepicker.js]: Are these changes related to your PR?

Comment (hahahana) [inline react-datepicker.js]: It's probably just my build setup? I haven't really been checking the output file. Can you rerun it on your machine?

Comment (RSO) [inline react-datepicker.js]: `master` has moved from React `0.11` to `0.12`, if you run `npm install` and try to rebuild this should all be back to normal!

**Comment by RSO** [comment]
Sorry for the late reply, I was hoping to reply sooner, but my flight to SF took a little more preparation than expected.

I'm kind of worried about the three inline if's that power this pull request. The logic feels a little implicit. I couldn't find a better spot/way to do this, so I guess we should just do it like this!

Did you try any other solutions to achieve this goal?

**Comment by hahahana** [comment]
Take a look at my second commit, I pulled in a method that I was saving in another PR, where it proved to really be necessary. I added a safeDateFormat method too, does that make it feel a bit less implicit?

What do you think? I personally can't think of anything else to do at this point.

**Comment by RSO** [comment]
Yeah, this is way better!
